### PR TITLE
1055 test cov green

### LINKED ISSUE #1055: Increase Testing Coverage to 90% for 0.7.0 Release
Goal is to get that coveted green code coverage badge!  

This is coverage goal currently in devel branch on April 1st.  

image

Increasing coverage for any of the above 15 files could help bump us up to 90%.

### PR DESCRIPTION
**Pull request by bms63**
Closes #1055

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/) 
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiral/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiral/articles/programming_strategy.html#deprecation-1)?
- [x] Update to all relevant roxygen headers and examples 
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md`
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue so that it closes after successful merging. 
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done!  Much love to your accomplishment!

### CODE REVIEW

**Comment by bms63** [comment]
Hi @thomas-neitmann @rossfarrugia  @bundfussr.  I updated a bunch of tests for covering warnings and errors - which helped to get us over 90%.  This is the bulk of these updates.  There are some S3 objects and Set Methods that I find hard to wrap my head around and create tests for, e.g. stuff in `derive_param_tte.R` and `iso_dtm.R`, but I think these should be covered and get us further along.   

I also cheated and brought the {rlang} tests for the `compat_friend_type.R` over so it gives us a little more bump.  

Do you want me to update anymore tests?  I might need a head with the S3 and Method stuff.  Hoping you guys don't mind reviewing this PR too! :)

Here is where we stand right now:

image

**Comment by rossfarrugia** [comment]
@bms63 for me >90% is a great target and i wouldn't stress going further. i defer to the others for review as i trust their technical expertise far more than my own.

**Comment by bms63** [comment]
@millerg23  this is ready for your review.

File: tests/testthat/test-assertions.R
Comment (millerg23) [inline tests/testthat/test-assertions.R:12]: same comment as above, add function name to `test_that` title, and maybe Test number?

**Review by millerg23 — COMMENTED** [review]
All updates look good, just need to add function name to `test_that` title and perhaps add Test number? (I added comment about this for `test-assertions.R` but applies to most of the scripts that were updated. Would be good to be consistent across the unit tests.

**Comment by bms63** [comment]
I feel like we need some sort of styler add-in to update the tests.  Going to hold off from updating them by hand as I think we need some sort of dev-utility function to do this.   I did updates `test-assertions.R` #714

**Comment by millerg23** [comment]
> I feel like we need some sort of styler add-in to update the tests. Going to hold off from updating them by hand as I think we need some sort of dev-utility function to do this. I did updates `test-assertions.R` #714

OK no problem think that makes sense, Stefan had mentioned something similar. I think we should be consistent within a unit test file though, so if you added a new test and tests in that file included the function name, then so should the new test. Can leave out numbering the tests. I will look through the tests and see which ones need function name added. Does that seem reasonable?

**Comment by bms63** [comment]
@millerg23 So I changed the file `test_assertions.R` to `test-assertions.R`.  I think that is some of the confusion here.  I didn't re-write that test file

File: tests/testthat/test-call_derivation.R
Comment (millerg23) [inline tests/testthat/test-call_derivation.R:95]: Update `test_that("Error is thrown if ... has no arguments"` to test_that("call_derivation Error is thrown if ... has no arguments".
Function name should be added for each of the new tests added, as existing tests have function name in the test_thsat call.

File: tests/testthat/test-derive_vars_aage.R
Comment (millerg23) [inline tests/testthat/test-derive_vars_aage.R:193]: Update `test_that("An Error is issued if age_unit is missing"` to `test_that("derive_var_age_years - An Error is issued if age_unit is missing"` to be consistent with rest of file.

File: tests/testthat/test-derive_vars_last_dose.R
Comment (millerg23) [inline tests/testthat/test-derive_vars_last_dose.R:179]: Update `test_that("An error is issued if same variable is found in both input datasets "` to `test_that("derive_vars_last_dose An error is issued if same variable is found in both input datasets "` to be consistent with rest of file.

**Review by millerg23 — COMMENTED** [review]
Just a couple of files to be updated to add function name in test_that call, only when existing tests have done so, to maintain consistency within each unit test file. Hope thats ok @bms63

**Comment by millerg23** [comment]
@bms63 I added just 1 more minor comment for `call_derivation` test file. And `lint` checks failing.

**Review by millerg23 — APPROVED** [review]

**Comment by millerg23** [comment]
@bms63 I have approved just need checks to finish running. Feel free to merge if you see checks have finished before I do!

**Pull request merged**
